WordPress/gutenberg
# List View: Add right-click behaviour to open block settings dropdown, and add right click overrides editor setting

## What?


Fixes #40287

Adds in right-click behaviour to the individual block buttons within the List View, that opens up the block settings dropdown menu instead of the browser default rick-click behaviour.

Because overriding browser default behaviour can be a contentious issue, this PR also adds in an editor setting (in the post and site editors) `allowRightClickOverrides` that defaults to `true`. That way, for folks who'd like to retain the browser default behaviour, they can opt-out.

## Why?


As raised in #40287, in other web apps that people are used to, right-clicking an item opens up a contextual menu that is part of the app — and for those spending a lot of time in the List View this change should make it a little easier to quickly make changes, rather than having to hover over the ellipsis menu to get to the settings dropdown.

## How?


* Add an `allowRightClickOverrides` setting in the post and site editors.
* At the button level in the List View, add an `onContextMenu` handler, and if the settings allow it, simulate a click on the settings dropdown when the button is right-clicked.

## Testing Instructions





1. In the post editor, add some blocks.
2. Open up the list view, and right click on an item in the list. The block settings menu should open up.
3. Multi-select multiple blocks and right click on an item — you should still be able to remove multiple blocks via this dropdown.
4. Open up the Preferences in the editor, and toggle the right click overrides setting off.
5. Check that the browser default context menu opens up when right clicking list view items.
6. Repeat the above in the site editor

## Screenshots or screencast 

| Right click on list item | The new setting |
| --- | --- |
| ![2022-05-13 13 37 03](https://user-images.githubusercontent.com/14988353/168206608-7dc38ba3-a094-4c42-afa9-473c55c73f09.gif) | <img width="495" alt="image" src="https://user-images.githubusercontent.com/14988353/168206657-9fe3afa2-9ae8-4a33-911f-c90c3ec5a57e.png"> |

Screengrab of right-click and toggling the setting:

![2022-05-13 13 41 22](https://user-images.githubusercontent.com/14988353/168207017-0c166986-c2a3-4615-84bf-e61d5b692230.gif)
